.github/workflows/ci.yml

```diff
@@ -260,3 +260,77 @@ jobs:
       # when imported in an NPM node_modules tree.
       - name: Prepack packages
         run: yarn lerna run --reject-cycles --concurrency 1 prepack
+
+  test-xs:
+    name: test-xs
+
+    # begin macro
+
+    runs-on: ${{ matrix.platform }}
+    strategy:
+      fail-fast: false
+      matrix:
+        node-version: [20.x]
+        # may include release tags and hashes:
+        moddable-version: [5.0.0]
+        platform: [ubuntu-latest]
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      # without this, setup-node errors on mismatched yarn versions
+      - run: corepack enable
+
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: yarn
+
+      - name: Install dependencies
+        run: yarn install --immutable
+
+      # end macro
+
+      - name: Run yarn build
+        run: yarn build
+
+      - name: Choose download or build
+        id: check-release
```

Comment:
Very much entertaining suggestions. If you have a suggestion that would parameterize the architecture, that’ll make it easier to turn on darwin and the holes-in-sides-of-houses variants in CI maybe someday.

Reply:
[There are](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs#runner-context), but not sure which are effectively needed, since GH cache is supposed to be different by default based on the OS (see [`enableCrossOsArchive`](https://github.com/actions/cache?tab=readme-ov-file#inputs)). Here is an [example in agoric-sdk](https://github.com/Agoric/agoric-sdk/blob/fd12d05b0f2242236e8526c4a4f4c3496b9c2e06/.github/actions/restore-node/action.yml#L134):
```
${{ runner.os }}-${{ runner.arch }}
```

A relevant PR/commit might be https://github.com/Agoric/agoric-sdk/pull/5512/commits/c9222a9d3c01b96c12d5b27c602e5c6a7ea088fc

Reply:
For the record, I’ve got a good idea of how to add platforms to the matrix and am electing not to at this juncture.